commons/impl/src/java/org/sakaiproject/commons/impl/CommonsSecurityManagerImpl.java

```diff
@@ -95,8 +97,12 @@ public boolean canCurrentUserEditPost(Post post) {
         String currentUser = sakaiProxy.getCurrentUserId();
 
         // If the current user is authenticated and the post author, yes.
-        if (currentUser != null && currentUser.equals(post.getCreatorId()) && sakaiProxy.isAllowedFunction(CommonsFunctions.POST_UPDATE_OWN, post.getSiteId())) {
-            return true;
+        if (currentUser != null && currentUser.equals(post.getCreatorId())) {
+            if (StringUtils.isEmpty(post.getId())) {
+                return sakaiProxy.isAllowedFunction(CommonsFunctions.POST_CREATE, post.getSiteId());
+            } else {
+                return sakaiProxy.isAllowedFunction(CommonsFunctions.POST_UPDATE_OWN, post.getSiteId());
```

Comment:
When does POST_UPDATE_ANY get checked?

Reply:
In the UI. I'll add it to the else statement. Hang on.

Reply:
It's there already, at the top of canCurrentUserEditPost